### ISSUE TITLE
Unable to generate Proof Request QR Code

### ISSUE THREAD

**Issue by cvarjao** [openwallet-foundation/bifold-wallet#700]
If the mediator is not correctly set up or connected, the screen becomes stuck while generating a QR code.

One of our developers experienced this issue when the mediator was down. The app started and functioned properly, but encountered a problem when attempting to generate a QR code for a proof, as seen in the video below:

<img width="321" alt="Screenshot 2023-04-05 at 10 46 31 AM" src="https://user-images.githubusercontent.com/1348549/230162195-3622ad8f-8d01-41dc-bfac-045962bd791f.png">

Although not verified, it is likely that you can reproduce this issue by either not providing a mediator URL or supplying an incorrect value.

**Comment by cvarjao** [comment]
@Artemkaaas, is that something we can check if the mediator is available?

**Comment by Artemkaaas** [comment]
I could not reproduce the infinity loader.  
I tried the following scenario: I turned off the internet after the app opened and tried to generate a proof request. 
Actual behavior I experienced: loader went away but there was neither a QR code nor an error message. 

In the PR added a showing of an error dialog if an error were thrown during proof request generation:  https://github.com/hyperledger/aries-mobile-agent-react-native/pull/707
